src/openshift/openshiftItem.ts

```diff
@@ -32,16 +32,42 @@ export abstract class OpenShiftItem {
         return new Promise((res) => setTimeout(res, timeout));
     }
 
+    static validateUniqueName(data: Array<OpenShiftObject>, value: string) {
+        const openshiftObject =  data.find((openshiftObject) =>  openshiftObject.getName() === value);
+        return openshiftObject && `This name is already used, please enter different name.`;
+    }
+
+    static async getName(message: string, data: Array<OpenShiftObject>) {
+        return await window.showInputBox({
+            prompt: `Please provide an ${message}`,
```

Comment:
Looks good, however, the sentence becomes grammatically incorrect once ${message} starts with a consonant (i.e. 'provide an component').